packages/eslint-plugin-svelte/src/rules/consistent-selector-style.ts

```diff
@@ -0,0 +1,286 @@
+		node.type === 'root' ||
+		node.type === 'selector'
+	) {
+		node.nodes.flatMap((node) => checkSelector(node, ruleGlobals));
+	}
+}
+
+/**
+ * Checks a class selector
+ */
+function checkClassSelector(node: SelectorClass, ruleGlobals: RuleGlobals): void {
+	const selection = ruleGlobals.classSelections.get(node.value) ?? [];
+	for (const styleValue of ruleGlobals.style) {
+		if (styleValue === 'class') {
+			return;
+		}
+		if (styleValue === 'id' && couldBeId(selection)) {
+			ruleGlobals.context.report({
+				messageId: 'classShouldBeId',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'type' && couldBeType(selection, ruleGlobals.typeSelections)) {
+			ruleGlobals.context.report({
+				messageId: 'classShouldBeType',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+	}
+}
+
+/**
+ * Checks an ID selector
+ */
+function checkIdSelector(node: SelectorIdentifier, ruleGlobals: RuleGlobals): void {
+	const selection = ruleGlobals.idSelections.get(node.value) ?? [];
+	for (const styleValue of ruleGlobals.style) {
+		if (styleValue === 'class') {
+			ruleGlobals.context.report({
+				messageId: 'idShouldBeClass',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'id') {
+			return;
+		}
+		if (styleValue === 'type' && couldBeType(selection, ruleGlobals.typeSelections)) {
+			ruleGlobals.context.report({
+				messageId: 'idShouldBeType',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+	}
+}
+
+/**
+ * Checks a type selector
+ */
+function checkTypeSelector(node: SelectorTag, ruleGlobals: RuleGlobals): void {
+	const selection = ruleGlobals.typeSelections.get(node.value) ?? [];
+	for (const styleValue of ruleGlobals.style) {
+		if (styleValue === 'class') {
+			ruleGlobals.context.report({
+				messageId: 'typeShouldBeClass',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'id' && couldBeId(selection)) {
+			ruleGlobals.context.report({
+				messageId: 'typeShouldBeId',
+				loc: ruleGlobals.parserServices.styleSelectorNodeLoc(node) as AST.SourceLocation
+			});
+			return;
+		}
+		if (styleValue === 'type') {
+			return;
+		}
+	}
+}
+
+/**
+ * Checks whether a given selection could be obtained using an ID selector
+ */
+function couldBeId(selection: AST.SvelteHTMLElement[]): boolean {
+	return selection.length <= 1;
+}
+
+/**
+ * Checks whether a given selection could be obtained using a type selector
+ */
+function couldBeType(
+	selection: AST.SvelteHTMLElement[],
+	typeSelections: Map<string, AST.SvelteHTMLElement[]>
+): boolean {
+	const types = new Set(selection.map((node) => node.name.name));
+	if (types.size > 1) {
```

Comment:
This doesn't report any selection which could be replaced by multiple type selections (e.g. replacing `.emphasized {}` by `i, b {}`). That can be added later I think (maybe gated with a configuration option?)